src/core/dbus-service.c

```diff
@@ -218,6 +219,72 @@ int bus_service_method_mount_image(sd_bus_message *message, void *userdata, sd_b
         return bus_service_method_mount(message, userdata, error, true);
 }
 
+int bus_service_method_dump_file_descriptor_store(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        Service *s = ASSERT_PTR(userdata);
+        int r;
+
+        assert(message);
+
+        r = mac_selinux_unit_access_check(UNIT(s), message, "status", error);
```

Comment:
I wonder if "status" isn't too lax for this. After all, this information used to be private. Are we showing only metadata about the FDs, or is there any way for an unprivileged user to use it to derive information about the content of the FDs too?

Reply:
i am not aware of any more appropriate selinux action for this. we can't just make up our own. the thing is that selinux policy upstream never responds to our requests about reworking the action names, hence I doubt we should innovate here.

Reply:
note that "systemd-analyze dump" is protected by the same selinux verb. We don't need to be more catholic than the pope on this hence.

Reply:
i think it's fine to list the file names. It would be problem if we'd show the contents of the files though

Reply:
> note that "systemd-analyze dump" is protected by the same selinux verb. We don't need to be more catholic than the pope on this hence.

Does that show the file names and other metadata associated with stored FDs?

> i am not aware of any more appropriate selinux action for this. we can't just make up our own. the thing is that selinux policy upstream never responds to our requests about reworking the action names, hence I doubt we should innovate here.

I can worry about that part, if needed. If we are saying that the filenames and the other things showed here are not privileged that's fine, but to be extra clear it would be worth to mention in the fdstore manpage that unprivileged users will generally be able to see that info. We don't want anybody to be surprised.

Reply:
the generic dbus dump calls do not show fdstore contnets right now. But we should change that. They should. see other comments about that.

I'll update the man page to make clear that contents of fds int he fdstore is considered sensitive, but not the fd metadata.